dasharo-security/measured-boot.robot

```diff
@@ -18,25 +18,376 @@ Resource            ../keys.robot
+    ${new_usb_storage_state}=    Evaluate    not ${usb_storage_state}
+    Set Option State    ${menu}    Enable USB Mass Storage    ${new_usb_storage_state}
+    Save Changes And Reset
+
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    FOR    ${pcr_hash}    IN    @{default_hashes}
+        ${pcr}    ${hash}=    Split String    ${pcr_hash}    separator=:
+        ${new_hash}=    Execute Command In Terminal    cat ${pcr}
+        IF    ${{'${pcr}'.endswith('/1')}}
+            Should Not Be Equal    ${hash}    ${new_hash}
+        ELSE
+            Should Be Equal    ${hash}    ${new_hash}
+        END
+    END
+
+MBO004.003 Changing Dasharo APU settings changes only PCR-1
+    [Documentation]    Check if changes to Dasharo APU settings influence PCR-1
+    ...    value and only PCR-1
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    Tests in firmware are not supported
+    Skip If    not ${APU_CONFIGURATION_MENU_SUPPORT}    Tests in Dasharo APU Menu are not supported
+    Power On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    @{default_hashes}=    Get PCRs State From Linux
+
+    Power On
+    ${menu}=    Enter Setup Menu Tianocore And Return Construction
+    ${menu}=    Enter Dasharo APU Configuration    ${menu}
+    ${core_boost_state}=    Get Option State    ${menu}    Core Performance Boost
+    ${new_core_boost_state}=    Evaluate    not ${core_boost_state}
+    Set Option State    ${menu}    Core Performance Boost    ${new_core_boost_state}
+    Save Changes And Reset
+
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    FOR    ${pcr_hash}    IN    @{default_hashes}
+        ${pcr}    ${hash}=    Split String    ${pcr_hash}    separator=:
+        ${new_hash}=    Execute Command In Terminal    cat ${pcr}
+        IF    ${{'${pcr}'.endswith('/1')}}
+            Should Not Be Equal    ${hash}    ${new_hash}
+        ELSE
+            Should Be Equal    ${hash}    ${new_hash}
+        END
+    END
+
+MBO005.001 Flashing firmware and reset to defaults results in same measurement
+    [Documentation]    Reset to defaults results in the same measurements as the
+    ...    one done after flashing
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    Tests in firmware are not supported
+    Skip If    not ${RESET_TO_DEFAULTS_SUPPORT}    Tests with "Reset to defaults" are not supported
+    ${fw}=    Get Variable Value    $FW_FILE    ${EMPTY}
+    Skip If    "${fw}" == "${EMPTY}"    No firmware to flash
+    Flash Firmware    ${FW_FILE}
+
+    Power Cycle On
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
+    ${default_pcr_state}=    Get PCRs State From Linux
+
+    Restore Secure Boot Defaults
+    Reset To Defaults Tianocore
+    Save Changes And Reset
+
+    Boot System Or From Connected Disk    ubuntu
+    Login To Linux
+    Switch To Root User
```

Comment:
This group of lines repeats a lot, a separate keyword for it will noticeably shorten the tests.

Reply:
fixed in [c458e30](https://github.com/Dasharo/open-source-firmware-validation/pull/326/commits/c458e304766e00bd12c3bca95bcac0d918229bb7), you are right there was a lot of repetition.